internal/test/wsu/wsu_test.go

```diff
@@ -44,6 +39,38 @@ var (
 	ubi8Image = "registry.access.redhat.com/ubi8/ubi:latest"
 )
 
+type WsuFramework struct {
+	framework *e2ef.TestFramework
+}
+
+// Setup initializes the framework. 'vmCountWithBuildWmcb' indicates how many VMs out of 'vmCount' created would run WSU
+// that uses built version of WMCB. Others will download the latest released version of WMCB as per the cluster version
+func (f *WsuFramework) Setup(vmCount int, vmCountWithBuiltWmcb int, credentials []*types.Credentials, skipVMsetup bool) error {
+	// framework holds the instantiation of test suite being executed. As of now, temp dir is hardcoded.
+	f.framework = &e2ef.TestFramework{}
+	// vmCountWithBuiltWmcb should be a subset of vmCount
+	if vmCountWithBuiltWmcb > vmCount {
+		return fmt.Errorf("count of VMs to use built WMCB should not be more than total count of VMs")
+	}
+	err := f.framework.Setup(vmCount, credentials, skipVMsetup)
+	if err != nil {
+		fmt.Errorf("framework setup failed: %v", err)
+	}
+	// Set 'buildWmcb' property of Windows VM
+	for i := 0; i < vmCountWithBuiltWmcb; i++ {
+		f.framework.WinVMs[i].SetBuildWmcb(true)
+	}
+	return nil
+}
+
+func (f *WsuFramework) RetrieveArtifacts() {
+	f.framework.RetrieveArtifacts()
+}
+
+func (f *WsuFramework) TearDown() {
+	f.framework.TearDown()
+}
+
```

Comment:
If you embed the framework, there will be no need of these wrappers unless we want to specialize the retrieval and tear down.

Reply:
We sort of discussed about this, @suhanime has some ideas around getting log files specific to wsu and wmcb runs.

Reply:
Yes - we can specialize the RetrieveArtifacts for WsuFramework when [WINC-155](https://issues.redhat.com/browse/WINC-155) is picked up